src/llmcompressor/transformers/sparsification/compressed_tensors_utils.py

```diff
@@ -114,14 +144,15 @@ def save_pretrained_wrapper(
                 return
 
             # if we've gotten to this point we have a config so we can run compression
-            kwargs["safe_serialization"] = True
+            kwargs["safe_serialization"] = kwargs.get("safe_serialization", True)
             if state_dict is None:
                 state_dict = get_state_dict_offloaded_model(model)
 
             # make sure we're on the main process when saving
             if state_dict is not None and len(state_dict) > 0:
                 compressed_state_dict = compressor.compress(model, state_dict)
                 kwargs["state_dict"] = compressed_state_dict
+                compressed_state_dict, _ = cleanse_shard(compressed_state_dict)
```

Comment:
This works but is unclear. `cleanse_shard` modifies `compressed_state_dict` through side effects, but also returns the modified value. It also means that since the `force_contiguous` option is done out-of-place, the changed value won't be saved in `kwargs["state_dict"]`

Reply:
Really the added line should be on line 154